### DISCUSSION TITLE
Fonts revert to default if disconnect or connect to another external monitor

### DISCUSSION THREAD

**Discussion by cxw620** [Tatsu-syo/noMeiryoUI#86] [Q&A] [answered]
Latest version of noMeiryoUI, Windows 10 22H2 Build 19045.3448. Laptop. 

After successfully setting fonts, the existing settings will not be affected whether I restart or just shut down my laptop. However, if I disconnect or connect to another external monitor, the fonts revert to default. 

I wonder if it's a bug, or a *feature* affected by Windows itself. Thanks.

**Comment by Tatsu-syo** [comment]
If laptop monitor's DPI and external monitor's DPI are different, Windows may changes system fonts.
I now changed my single monitor's resolution and DPI 4K and 144DPI to 2K & 96DPI, system fonts are changed.
Monitor disconnection become trigger to change system fonts for Windows.
How about?

**Reply by Tatsu-syo** [reply]
Oh, I forgot to write testing environments.
I tested on Windows 10 22H2 Build 19045.3448 and 4K monitor.

**Reply by cxw620** [reply]
My test result is the same as yours. 
So can we conclude that noMeiryoUI itself just changes current system fonts, and we can only set fonts again every time the system fonts are changed to the default value due to **DPI changes** or **monitor connection status changes**?

**Reply by Tatsu-syo** [reply, marked as answer]
Probably Yes. It seems system fonts are changed to the default value due to DPI changes or monitor connection status changes/
